boot/bootutil/src/loader.c

```diff
@@ -703,14 +712,32 @@ boot_validated_swap_type(struct boot_loader_state *state,
     swap_type = boot_swap_type_multi(BOOT_CURR_IMG(state));
     if (BOOT_IS_UPGRADE(swap_type)) {
         /* Boot loader wants to switch to the secondary slot.
-         * Ensure image is valid.
+         * Ensure image is valid. This checks header, hash/signature and trailer
          */
         rc = boot_validate_slot(state, BOOT_SECONDARY_SLOT, bs);
+	/* Check for network update type */
         if (rc == 1) {
             swap_type = BOOT_SWAP_TYPE_NONE;
         } else if (rc != 0) {
             swap_type = BOOT_SWAP_TYPE_FAIL;
         }
+
+	vtable_addr = (uint32_t)hdr + hdr->ih_hdr_size;
+	vtable = (uint32_t *)(vtable_addr);
+	reset_addr = vtable[1];
+	if (reset_addr > PM_nrf5340pdk_nrf5340_cpunet_B0N_ADDRESS) {
```

Comment:
Shouldn't this be `ifdef`-ed in only for nrf53cpuapp?

Reply:
And we should check that `reset_addr == PM_nrf5340pdk_nrf5340_cpunet_B0N_ADDRESS`

Reply:
Note that we will change this to the proper domain name once https://github.com/nrfconnect/sdk-nrf/pull/2527 is in

Reply:
reset table addr should not be b0n address since it's the start addr of app

Reply:
Keeping this there, are we missing a check?